app/src/main/java/com/mymeetings/android/view/widgets/CalendarEventWidgetService.kt

```diff
@@ -14,49 +16,53 @@ import androidx.lifecycle.LifecycleOwner
 import androidx.lifecycle.LifecycleRegistry
 import androidx.lifecycle.Observer
 import com.mymeetings.android.R
+import com.mymeetings.android.model.AlertType
 import com.mymeetings.android.model.CalendarEvent
-import com.mymeetings.android.view.activities.ui.home.MeetingsViewModel
+import com.mymeetings.android.model.CalendarEventAlertManager
+import com.mymeetings.android.utils.ClockUtils
+import com.mymeetings.android.view.activities.ui.home.CalendarEventsViewModel
 import org.koin.android.ext.android.get
 
 
-class MeetingAppWidgetService : RemoteViewsService() {
+class CalendarEventWidgetService : RemoteViewsService() {
 
     override fun onGetViewFactory(intent: Intent?): RemoteViewsFactory {
-        return get<MeetingWidgetRemoteViewFactory>()
+        return get<CalendarEventWidgetRemoteViewFactory>()
     }
 }
 
-class MeetingWidgetRemoteViewFactory(
+class CalendarEventWidgetRemoteViewFactory(
     private val context: Context,
-    private val meetingsViewModel: MeetingsViewModel
+    private val calendarEventsViewModel: CalendarEventsViewModel,
+    private val calendarEventAlertManager: CalendarEventAlertManager,
+    private val clockUtils: ClockUtils
 ) : RemoteViewsService.RemoteViewsFactory, LifecycleOwner {
 
-    private var calendarEvents : List<CalendarEvent>? = null
+    private var calendarEventsWithAlertType : List<Pair<CalendarEvent, AlertType>>? = null
 
     private val lifecycleDispatcher: WidgetServiceLifecycleDispatcher =
         WidgetServiceLifecycleDispatcher(this)
 
 
     override fun onCreate() {
         lifecycleDispatcher.onConstructor()
-        meetingsViewModel.meetingsLiveData.observe(this, Observer {
-            calendarEvents = it
+        calendarEventsViewModel.getCalendarEventLiveData().observe(this, Observer {
+            calendarEventsWithAlertType = calendarEventAlertManager.getCalendarEventAlerts(it)
             val appWidgetManager = AppWidgetManager.getInstance(context)
             val appWidgetIds = appWidgetManager.getAppWidgetIds(
-                ComponentName(context, MeetingAppWidgetProvider::class.java)
+                ComponentName(context, CalendarEventsWidgetProvider::class.java)
             )
             appWidgetManager.notifyAppWidgetViewDataChanged(appWidgetIds, R.id.listView)
-            //TODO refresh this adapter.
         })
-        meetingsViewModel.syncEvents()
+        calendarEventsViewModel.syncEvents()
     }
 
     override fun getLoadingView(): RemoteViews {
         return RemoteViews(context.packageName, R.layout.item_meeting)
     }
 
     override fun getItemId(position: Int): Long {
-        return calendarEvents?.get(position)?.id?:position.toLong()
+        return calendarEventsWithAlertType?.get(position)?.first?.id?:position.toLong()
```

Comment:
When we use a model instead of `Pair`, this line of code will become readable as we will end up using `calendarEvent` instead of `first`.

Reply:
Changed Pair<CalendarEvent, AlertType> to CalendarEventWithViewAlert